cmd/krel/cmd/ff.go

```diff
@@ -0,0 +1,307 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+
+	"k8s.io/release/pkg/util"
+)
+
+var cfgFile string
+
+const defaultMasterRef string = "HEAD"
+
+type Options struct {
+	branch    string
+	masterRef string
+	org       string
+	nomock    bool
+	cleanup   bool
+}
+
+var opts = &Options{}
+
+// ffCmd represents the base command when called without any subcommands
+var ffCmd = &cobra.Command{
+	Use:   "ff --branch <release-branch> [--ref <master-ref>] [--nomock] [--cleanup]",
+	Short: "ff fast forwards a Kubernetes release branch",
+	Long: `ff fast forwards a branch to a specified master object reference
+(defaults to HEAD), and then prepares the branch as a Kubernetes release branch:
+
+- Run hack/update-all.sh to ensure compliance of generated files`,
+	Example: "krel ff --branch release-1.16 39d0135e --ref HEAD --cleanup",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if err := runFf(opts); err != nil {
+			return err
+		}
+
+		return nil
+	},
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	ffCmd.PersistentFlags().StringVar(&opts.branch, "branch", "", "branch")
+	ffCmd.PersistentFlags().StringVar(&opts.masterRef, "ref", defaultMasterRef, "ref on master")
+	ffCmd.PersistentFlags().StringVar(&opts.org, "org", util.DefaultGithubOrg, "org to run tool against")
+
+	rootCmd.AddCommand(ffCmd)
+}
+
+func runFf(opts *Options) error {
+	// TODO: Add usage/help
+	// TODO: Check prerequisites (git, jq, go, make)
+	// TODO: Set positional args
+	// TODO: Fail on empty branch
+	// TODO: Fail on GITHUB_TOKEN not set
+
+	branch := opts.branch
+	masterRef := opts.masterRef
+	remote := util.DefaultRemote
+	remoteMaster := fmt.Sprintf("%s/%s", remote, "master")
+
+	log.Printf("Preparing to fast-forward master@%s onto the %s branch...\n", masterRef, branch)
+
+	nomock := opts.nomock
+	dryRunFlag := "--dry-run"
+	if nomock {
+		// TODO: Set this to empty string once we're ready to turn on the tool
+		log.Println("Running in no-mock mode!")
+		dryRunFlag = "--dry-run"
+	}
+
+	re := regexp.MustCompile(util.BranchRE)
```

Comment:
I would add this as a stand alone function to the utils and avoid making `BranchRE` public, like: `util.IsReleaseBranch(string)`